src/utils/components/SSHAccess/utils.ts

```diff
@@ -44,44 +46,60 @@ export const deleteSSHService = (sshService: IoK8sApiCoreV1Service) =>
     ns: sshService?.metadata?.namespace,
   });
 
-export const createSSHService = (
+export const addSSHSelectorLabelToVM = async (
+  vm: V1VirtualMachine,
+  vmi: V1VirtualMachineInstance,
+  labelValue,
+) => {
+  const vmWithLabel = produce(vm, (draftVM) => {
+    if (!draftVM?.spec?.template?.metadata) {
+      draftVM.spec.template.metadata = {};
```

Comment:
Maybe you could use `ensurePath` here, the one from the other PR? :)

Reply:
fixed